app/models/manageiq/providers/redhat/infra_manager/refresh_parser.rb

```diff
@@ -334,9 +340,28 @@ def self.host_inv_to_network_hashes(inv, guest_device_uids)
   def self.host_inv_to_os_hash(inv, hostname)
     return nil if inv.nil?
 
-    result = {:name => hostname}
-    result[:product_name] = 'linux'
-    result
+    {
+      :name         => hostname,
+      :product_type => 'linux',
+      :product_name => extract_host_os_name(inv),
+      :version      => extract_host_os_full_version(inv[:os])
+    }
+  end
+
+  def self.extract_host_os_full_version(host_os)
```

Comment:
I modified a bit the method to make it clearer: The `inv[:os]` is checked anyway in the `extract_host_os_full_version` method not to be `nil`. The check for it on `extract_host_os_name` was done for safely extracting of `host_inv[:os][:type]`, not as a precaution for `extract_host_os_name`.